### ISSUE TITLE
Unable to createServerSupabaseClient with different schema

### ISSUE THREAD

**Issue by adamzerner** [supabase/supabase#11931]
# Bug report

## Describe the bug

To use a different schema for the supabase client, you're [supposed to](https://supabase.com/docs/reference/javascript/initializing) do something like this:

```
const supabase = createClient('https://xyzcompany.supabase.co', 'public-anon-key', {
  db: { schema: 'other_schema' }
})
```

I'm not seeing a way to do that though with `createServerSupabaseClient`. I'm using TypeScript. When I do this:

```
const supabase = createServerSupabaseClient(
  { req, res },
  process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY,
  {
    db: {
      schema: "auth",
    },
  }
);
```
I get this error:

```
Expected 1-2 arguments, but got 3.
```

And when I do this:

```
const supabase = createServerSupabaseClient(
  { req, res },
  {
    db: {
      schema: "auth",
    },
  }
);
```

I get this error:

```
Argument of type '{ db: { schema: string; }; }' is not assignable to parameter of type '{ cookieOptions?: CookieOptions | undefined; }'.
  Object literal may only specify known properties, and 'db' does not exist in type '{ cookieOptions?: CookieOptions | undefined; }'.
```

I don't see anything on the [Supabase Auth with Next.js](https://supabase.com/docs/guides/auth/auth-helpers/nextjs) page about this and don't see any other relevant docs.

## To Reproduce

Not really applicable.

## Expected behavior

1) The ability to call `createServerSupabaseClient` with the schema you want.

2) Documentation for that ability.

## Screenshots

n/a

## System information

- OS: n/a
- Browser (if applies): n/a
- Version of supabase-js: see below
- Version of Node.js: 16.11.1

```
"@supabase/auth-helpers-nextjs": "^0.5.2",
"@supabase/auth-helpers-react": "^0.3.1",
"@supabase/supabase-js": "^2.4.0",
```

## Additional context

No

**Comment by ruggi99** [comment]
Hi @adamzerner 
I've read the documentation and in fact it is not clear how to pass additional options to `createServerSupabaseClient`.
Looking at source code: https://github.com/supabase/auth-helpers/blob/1c07049b29b0dfed414264e31350d3f2224718de/packages/nextjs/src/index.ts#L67
you can see that you should pass a object as the second parameter and one field is `options`.
So you should end up with this if I'm not wrong:
```js
const supabase = createServerSupabaseClient(
  { req, res },
  {
    options: {
      db: {
        schema: "auth",
      },
    },
  }
);
```

**Comment by adamzerner** [comment]
Hey @ruggi99 , thanks for the reply. Looking at the source code it does look to me like that should work. However:

1: When I try it it doesn't work.

```
const supabase = createServerSupabaseClient(
  { req, res },
  {
    // @ts-expect-error
    options: {
      db: {
        schema: "auth",
      },
    },
  }
);
console.log(supabase);
```
With that code I see that `supabase.rest.schema` is `'public'` not `'auth'`. And the follow up code of this doesn't succeed.
```
await supabase
  .from("users")
  .update({
    raw_user_meta_data: {
      role: "paid",
      customerId: checkoutSession.customer,
      subscriptionId: checkoutSession.subscription,
    },
  })
  .eq("email", email);
```

2: I get a type error. Without the `@ts-expect-error`:

```
Argument of type '{ options: { db: { schema: string; }; }; }' is not assignable to parameter of type '{ cookieOptions?: CookieOptions | undefined; }'.
  Object literal may only specify known properties, and 'options' does not exist in type '{ cookieOptions?: CookieOptions | undefined; }'.ts(2345)
```

**Comment by ruggi99** [comment]
@adamzerner about a week ago they changed the behavior and allowed to pass options.
Ensure you are on the latest version of `@supabase/auth-helpers-nextjs` (`0.5.4` as now) and it should work)

**Comment by adamzerner** [comment]
@ruggi99 Ah ok. After upgrading to `0.5.4` I don't get the type errors anymore but now I get a "The schema must be one of the following: public, storage, graphql_public" error when I run the following code:

```
const { data, error } = await supabase
  .from("users")
  .update({
    raw_user_meta_data: {
      role: "paid",
      customerId: checkoutSession.customer,
      subscriptionId: checkoutSession.subscription,
    },
  })
  .eq("email", email);
console.log(error);
```
Full error:
```
{
  code: 'PGRST106',
  details: null,
  hint: null,
  message: 'The schema must be one of the following: public, storage, graphql_public'
}
```
Is that expected? Is there something I can do to be able to access the `auth` schema? Note: I used the "User Management Starter" [recommended](https://supabase.com/docs/guides/getting-started/tutorials/with-nextjs#set-up-the-database-schema) in the Next.js guide.

**Comment by ruggi99** [comment]
@adamzerner it is expected because you are using the anon key and `auth` schema is not exposed for security reasons (anyone would be able to update users data programatically). Obviously exposing the schema is not an option nor it is recomended.
What you have to do is to use the service_key (be sure to never expose it) and use GoTrue Admin feature as described here:
https://supabase.com/docs/reference/javascript/auth-admin-updateuserbyid

**Comment by adamzerner** [comment]
I'm running this code on the server and have access to the `service_key`. Is there a way to use it in my current code? I don't have access to the users id when this code runs, it's coming from a Stripe webhook and what I have access to is their email.

**Comment by ruggi99** [comment]
I'm afraid you have to retrieve all users and find the user you are looking for.
This slows down the entire process but I have no other solutions in mind now

**Comment by TEMHITHORPHE** [comment]
@adamzerner 
Here you go mate, follow the answer to the same question on this "discussion" I had with @sweatybridge  a couple of weeks ago, you should be able to apply the same solution, and hope it works for you ... yea ... it actually did work for me.

PS: I also had to add the "auth" schema to my search path (which can be done under settings).
PS: I am definitely running all my code server side (exposing the service_role key is not pleasant).

**Comment by adamzerner** [comment]
@TEMHITHORPHE Thanks for the help but unfortunately that didn't work for me. I ran:

```
GRANT ALL ON TABLE auth.users TO service_role;
```

In the SQL Editor and got:

```
Success. No rows returned
```

but then I ended up with that same "The schema must be one of the following: public, storage, graphql_public" error. Would have been nice though.

@ruggi99 I see, thanks for the help. I think my original question has been answered. I am having [this issue](https://github.com/supabase/supabase/discussions/12010) now but it's unrelated.
